cpp/src/arrow/util/bit_util.h

```diff
@@ -1170,5 +1170,32 @@ class BitsetStack {
   std::vector<int> offsets_;
 };
 
+/// \brief A class that scans through a true/false bitmap to yield blocks of up
+/// to 256 bits at a time along with their popcount. This is used to accelerate
+/// processing of mostly-not-null array data.
+class ARROW_EXPORT BitBlockCounter {
+ public:
+  struct Block {
+    int16_t length;
```

Comment:
did you check if this affects benchmarks, I think popcount returns ints so there might be some miniscule advantage to keep it at a larger size.

Reply:
It didn't seem to affect them